Sources/PropertyEditor.swift

```diff
@@ -0,0 +1,680 @@
+	/// value whenever the setter is invoked.
+	///
+	/// If `success` is returned by `body`, the associated value would be
+	/// persisted to `inner`. Otherwise, the failure would be emitted by the
+	/// `validations` signal.
+	///
+	/// The validation results of `inner` would be propagated to the created
+	/// `PropertyEditor`.
+	///
+	/// - parameters:
+	///   - inner: The inner property to wrap.
+	///   - transform: The value transform for the presentation.
+	///   - errorTransform: The error transform for the presentation.
+	///   - body: The closure to invoke for any proposed value to `self`.
+	public convenience init<T: _PropertyEditorProtocol>(
+		_ inner: T,
+		transform: @escaping (T.Value) -> Value,
+		errorTransform: @escaping (T.ValidationError) -> ValidationError,
+		_ body: @escaping (Value) -> Result<T.Value, ValidationError>
+	) {
+		self.init(inner, transform: transform, validator: body, validationSetup: { validations in
+			validations <~ inner.property.result.signal
+				.map { $0.map(transform, errorTransform, validator: body) }
+		})
+	}
+
+	/// Create an `PropertyEditor` that presents `inner` as an
+	/// `PropertyEditor` of `U` value, and invoke `body` with the proposed
+	/// value whenever the setter is invoked.
+	///
+	/// If `success` is returned by `body`, the associated value would be
+	/// persisted to `inner`. Otherwise, the failure would be emitted by the
+	/// `validations` signal.
+	///
+	/// - parameters:
+	///   - inner: The inner property to wrap.
+	///   - transform: The value transform for the presentation.
+	///   - body: The closure to invoke for any proposed value to `self`.
+	public convenience init<T: _PropertyEditorProtocol>(
+		_ inner: T,
+		transform: @escaping (T.Value) -> Value,
+		_ body: @escaping (Value) -> Result<T.Value, ValidationError>
+	) where T.ValidationError == NoError {
+		self.init(inner, transform: transform, validator: body, validationSetup: nil)
+	}
+
+	private init<T: _PropertyEditorProtocol>(
+		_ inner: T,
+		transform: @escaping (T.Value) -> Value,
+		validator: @escaping (Value) -> Result<T.Value, ValidationError>,
+		validationSetup: ((MutableProperty<ValidationResult<Value, ValidationError>>) -> Void)?
+	) {
+		let _validations: MutableProperty<ValidationResult<Value, ValidationError>> = inner.property.rootBox.lock {
+			let property = MutableProperty(ValidationResult(inner.property.commited.value, transform: transform, validator: validator))
+			validationSetup?(property)
+			return property
+		}
+
+		self.result = Property(capturing: _validations)
+		self.lifetime = inner.property.lifetime
+		self.commited = inner.property.commited.map(transform)
+		self.rootBox = inner.property.rootBox
+
+
+		action = { input in
+			switch validator(input) {
+			case let .success(innerResult):
+				inner.property.action(innerResult)
+
+			case let .failure(error):
+				_validations.value = .failure(input, error)
+			}
+		}
+	}
+
+	@discardableResult
+	public func `try`(_ newValue: Value) {
```

Comment:
I think `attemptSet()`(or if you don't mind the verbosity, `attemptToSet(value newValue: Value)` might be more appropriate here. The same applies elsewhere—I think I prefer `attempt` to `try` seeing how this has little/nothing to do with Swift exceptions.

(My distaste for `try` here is exacerbated by the fact that you also have actual uses of Swift's `try` keyword in the same file.)